### PR TITLE
[TASK] Add an example on how to migrate via upgrade script from switc…

### PR DESCRIPTION
**Pull request by linawolf**
…hable controller actions

releases: main, 12.4

### CODE REVIEW

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Examples.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+..  _upgrade-wizard-examples:
+
+===================================
+Examples for common upgrade wizards
+===================================
+
+..  _upgrade-wizard-examples-switchable-controller-actions:
+
+Upgrade script to replace switchable controller actions
+=======================================================
```
Comment (froemken) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Examples.rst:11]:
```suggestion
Upgrade wizard to replace switchable controller actions
=======================================================
```

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+use TYPO3\CMS\Install\Updates\UpgradeWizardInterface;
+
+#[UpgradeWizard('myExtension_switchableControllerActionUpgradeWizard')]
+final class SwitchableControllerActionUpgradeWizard implements UpgradeWizardInterface
+{
+    private const TABLE = 'tt_content';
+    private const PLUGIN = 'myextension_myplugin';
+    public function __construct(
+        private readonly ConnectionPool $connectionPool,
+        private readonly FlexFormService $flexFormService
```
Comment (froemken) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php:20]: Unsure. Each lines will end with `,` now, right?

```suggestion
        private readonly FlexFormService $flexFormService,
```

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    ) {
+    }
+
+    public function executeUpdate(): bool
+    {
+        $result = 0;
+        $result += $this->migratePlugin(self::PLUGIN, 'Videoplayer->list', 'ableplayer_videoplayerlist');
```
Comment (froemken) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php:27]: `ableplayer_`? Maybe `myextension_`? Or something with `switchable_`?

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    public function __construct(
+        private readonly ConnectionPool $connectionPool,
+        private readonly FlexFormService $flexFormService,
+    ) {
+    }
+
+    public function executeUpdate(): bool
+    {
+        $result = 0;
+        $result += $this->migratePlugin(self::PLUGIN, 'Videoplayer->list', 'ableplayer_videoplayerlist');
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php:27]:
```suggestion
        $result += $this->migratePlugin(self::PLUGIN, 'Videoplayer->list', 'myextension_videoplayerlist');
```

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        private readonly ConnectionPool $connectionPool,
+        private readonly FlexFormService $flexFormService,
+    ) {
+    }
+
+    public function executeUpdate(): bool
+    {
+        $result = 0;
+        $result += $this->migratePlugin(self::PLUGIN, 'Videoplayer->list', 'myextension_videoplayerlist');
+        $result += $this->migratePlugin(self::PLUGIN, 'Videoplayer->overview;Videoplayer->detail', 'ableplayer_audioplayeroverviewdetail');
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php:28]:
```suggestion
        $result += $this->migratePlugin(self::PLUGIN, 'MyController->list', 'myextension_mycontrollerlist');
        $result += $this->migratePlugin(self::PLUGIN, 'MyController->overview;MyController->detail', 'myextension_mycontrolleroverviewdetail');
```

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php (hunk 3 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/_SwitchableControllerActionUpgradeWizard.php]: Thanks for catching this, fixed it

**Pull request merged**